src/_backend_agg_basic_types.h

```diff
@@ -121,4 +126,135 @@ class GCAgg
     GCAgg &operator=(const GCAgg &);
 };
 
+namespace PYBIND11_NAMESPACE { namespace detail {
+    template <> struct type_caster<agg::line_cap_e> {
+    public:
+        PYBIND11_TYPE_CASTER(agg::line_cap_e, const_name("line_cap_e"));
+
+        bool load(handle src, bool) {
+            const std::unordered_map<std::string, agg::line_cap_e> enum_values = {
+                {"butt", agg::butt_cap},
+                {"round", agg::round_cap},
+                {"projecting", agg::square_cap},
+            };
+            value = enum_values.at(src.cast<std::string>());
+            return true;
+        }
+    };
+
+    template <> struct type_caster<agg::line_join_e> {
+    public:
+        PYBIND11_TYPE_CASTER(agg::line_join_e, const_name("line_join_e"));
+
+        bool load(handle src, bool) {
+            const std::unordered_map<std::string, agg::line_join_e> enum_values = {
+                {"miter", agg::miter_join_revert},
+                {"round", agg::round_join},
+                {"bevel", agg::bevel_join},
+            };
+            value = agg::miter_join_revert;
```

Comment:
I don't think you need the default here (either the next line will throw or it won't)?

Reply:
There wasn't a default in `line_cap_e`, so this must have been some leftover that wasn't cleaned up.